tools/ci/provision_mac.sh

```diff
@@ -18,12 +18,14 @@ brew \
   --overwrite python@3.12 \
      clang-format@21 \
      fontconfig \
+     gcovr \
```

Comment:
I've added coverage dependencies to `provision_mac.sh`, and I don't think I've done anything more platform-dependent than existing code in `CMakeLists.txt` and `run_all_tests.sh`, but coverage on Mac is untested since the only Mac build in CI uses qmake. Feel free to let me know any thoughts on if I should replace or extend `.github/workflows/mac2023.yml` with a CMake build to run Mac coverage in CI.